### PR TITLE
Add documentation to RHS

### PR DESCRIPTION
**Pull request by Steven-Roberts**

### CODE REVIEW

File: toolbox/+otp/RHS.m [hunk 1]
```diff
@@ -1,36 +1,304 @@
 classdef RHS
+    % The right-hand side and related properties of a differential equation.
+    %
+    % This immutable class contains properties required by time integrators and other numerical methods to describe the
+    % following differential equation:
+    %
+    % $$M(t, y) y' = f(t, y)$$
+    %
+    % The mass matrix $M(t, y)$ is permitted to be singular in which case we have a differential-algebraic equation.
```
Comment (AndreyAPopov) [inline toolbox/+otp/RHS.m:9]: It seems like most of the documentation is written without the use of the mathematical We, so maybe do passive voice?

File: toolbox/+otp/RHS.m [hunk 2]
```diff
@@ -1,36 +1,304 @@
...
+    % ``RHS`` includes most of the properties available in
+    % `odeset <https://www.mathworks.com/help/matlab/ref/odeset.html>`_ so that it can easily be used with built-in ODE
+    % solvers.
+    %
+    % Examples
+    % --------
+    % >>> rhs = otp.RHS(@(t, y) [y(2) + t; y(1) * y(2)], ...
+    % ...     'Jacobian', @(~, y) [0, 1; y(2), y(1)], ...
+    % ...     'Mass', [1, 2; 3, 4]);
+    % >>> sol = ode23s(rhs.F, [0, 1], [1, -1], rhs.odeset());
```
Comment (AndreyAPopov) [inline toolbox/+otp/RHS.m:19]: Oh, I didn't know we supported rhs.odeset, nice!

File: toolbox/+otp/RHS.m [hunk 3]
```diff
@@ -1,36 +1,304 @@
...
+    % --------
+    % odeset : https://www.mathworks.com/help/matlab/ref/odeset.html
+
     properties (SetAccess = private)
+        % The function handle for $f$ in the differential equation $M(t, y) y' = f(t, y)$.
+        %
+        % Parameters
+        % ----------
+        % t : numeric
+        %    The time at which $f$ is evaluated.
```
Comment (AndreyAPopov) [inline toolbox/+otp/RHS.m:41]: if vectorized is on, t can and should be a vector, right?

File: toolbox/+otp/RHS.m [hunk 4]
```diff
@@ -1,36 +1,304 @@
...
+        %
+        % Returns
+        % -------
+        % terminal : logical
+        %    ``true`` if the integration should stop and ``false`` if it can continue.
+        % newProblem : Problem
+        %    A new Problem to use after the event, possibly with the time span, initial conditions, or parameters
+        %    updated.
+        %
+        % Warning
```
Comment (AndreyAPopov) [inline toolbox/+otp/RHS.m:145]: Is there a way to make this red/orange, or in a box? I feel like warnings should be highlighted in some way.

File: toolbox/+otp/RHS.m [hunk 5]
```diff
@@ -1,36 +1,304 @@
...
+        % If set, it is a function handle with the following signature:
+        %
+        % Parameters
+        % ----------
+        % t : numeric
+        %    The time at which the Hessian is evaluated.
+        % y : numeric(:, 1)
+        %    The state at which the Hessian is evaluated.
+        % u, v : numeric(:, 1)
+        %    A vector applied to the Hessian.
```
Comment (AndreyAPopov) [inline toolbox/+otp/RHS.m:241]: The vectors, plural

File: toolbox/+otp/RHS.m [hunk 6]
```diff
@@ -111,9 +379,20 @@
...
+            %    Additional name-value pairs which have precedence over the values in this class.
+            %
+            % Returns
+            % -------
+            % opts : struct
+            %    An options strucutre containing the subset of properties compatible with built-in ODE solvers.
+
             opts = odeset( ...
                 'Events', obj.Events, ...
-                'InitialSlope', obj.InitialSlope, ...
```
Comment (AndreyAPopov) [inline toolbox/+otp/RHS.m:116]: Why did we remove this?

File: toolbox/+otp/RHS.m (hunk 4 above)
Comment (Steven-Roberts) [reply toolbox/+otp/RHS.m:145]: Already is

File: toolbox/+otp/RHS.m (hunk 6 above)
Comment (Steven-Roberts) [reply toolbox/+otp/RHS.m:116]: That property is y0', so like y0, it belongs in the Problem not the RHS

File: toolbox/+otp/RHS.m (hunk 2 above)
Comment (Steven-Roberts) [reply toolbox/+otp/RHS.m:19]: Also works as `odeset(p.RHS)`

**Pull request merged**